### ISSUE TITLE
[help wanted] Opencore 0.6.4 with YogaSMC

### ISSUE THREAD

**Issue by xuandung38** [simprecicchiani/ThinkPad-T460s-macOS-OpenCore#71]
I find YogaSMC to have more superior features than ThinkpadAssistant. For example the Fan control, Charge Threshold Management ... really useful. Would you mind spending some of your precious time trying to research it?

**Comment by aptonline** [comment]
I second this, YogaSMC looks great, especially the fan control abilities. @xuandung38 my understanding is that this replaced VirtualSMC is this correct?

**Comment by xuandung38** [comment]
No! YogaSMC based on VirtualSMC. So we allway need VirtualSMC and load it before YogaSMC.

**Comment by xuandung38** [comment]
I tried YogaSMC, it works very well. But I don't know how to fix sleep so I'm still temporarily using this repo. If you need it I will send it to you.

**Comment by simprecicchiani** [comment]
Hi, I did some researches but unfortunately I don't have the knowledge and time to implement this right now.

**Comment by aptonline** [comment]
maybe @xuandung38 can point you in the right direction as what needs to be done? from what @xuandung38  is saying its a KEXT and System preferences pane that runs alongside VirtualSMC and can be used instead of ThinkAssistant so in essence its about adding the KEXT Into the EFI built with a supplementary step of adding the preference pane afterwards.

**Comment by xuandung38** [comment]
Here is my EFI OC 0.6.3 with yogaSMC. 
I just don't know how to fix sleep for it.

You can take it and researches

[T460SxYogaSMC.zip](https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/files/5674774/T460SxYogaSMC.zip)

**Comment by simprecicchiani** [comment]
There’s no need for supplementary SSDTs?

**Comment by xuandung38** [comment]
No, we need to add some additional SSDT to manage PIN, Fan and LED control.

**Comment by xuandung38** [comment]
Guide:

**Comment by xuandung38** [comment]
Here is guide:
Change ERAM in method RE1B to ERM2 or any.
Add 2 method to SSDT_BAT0 like this
![image](https://user-images.githubusercontent.com/8670315/101891458-a149b580-3bd4-11eb-8120-493f64c2d8d4.png)
Download [Latest release](https://github.com/zhen-zen/YogaSMC/releases/tag/1.3.0)
Compile and add SSD-DYTC-Tx80.dsl + SSD-THINK.dsl to ACPI

Remove all __QXX_ method in ACPI->patch

Add YogaSMC.kext to kext and kernel -> done

**Comment by simprecicchiani** [comment]
Have you already done that? Does it fix sleep issues?

**Comment by xuandung38** [comment]
When i use repatch your ACPI it work well. But led control not work , i'm researching.

![Screen Shot 2020-12-13 at 16 23 20](https://user-images.githubusercontent.com/8670315/102007916-85264f80-3d5f-11eb-9176-cc91243720fd.png)
[EFI.zip](https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/files/5684394/EFI.zip)

**Comment by xuandung38** [comment]
try it and tell me.

**Comment by simprecicchiani** [comment]
I made some changes to the original ACPI patches from OC-Little, then I looked into @tylernguyen x1c6 config to see how he implemented the tool.
I'm still not confident with ACPI but it seems that most of YogaSMC's functions are working.
Some LEDs are fully functional, others are partially supported (i.e. charging light can't pulse, only blink).
[EFI.zip](https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/files/5718398/EFI.zip)

**Comment by xuandung38** [comment]
> I made some changes to the original ACPI patches from OC-Little, then I looked into @tylernguyen x1c6 config to see how he implemented the tool.
> I'm still not confident with ACPI but it seems that most of YogaSMC's functions are working.
> Some LEDs are fully functional, others are partially supported (i.e. charging light can't pulse, only blink).
> [EFI.zip](https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/files/5718406/EFI.zip)
> 
> Edit: I also added BrightnessKeys.kext

Let's me test it!

**Comment by xuandung38** [comment]
Everythings is ok , but Charge Threshold function notwork.

**Comment by xuandung38** [comment]
Any update ?

**Comment by aptonline** [comment]
Progress on this is looking good, noticed new version available with a ton of improvements: https://github.com/zhen-zen/YogaSMC/releases/tag/1.4.0

**Comment by simprecicchiani** [comment]
> Progress on this is looking good, noticed new version available with a ton of improvements: https://github.com/zhen-zen/YogaSMC/releases/tag/1.4.0

Led control and Mic button broke with this release

**Comment by xuandung38** [comment]
Hi @simprecicchiani ! I have been using your efi so far I have not seen any inconvenience or bugs, I think it's time to release a separate branch for this version.

**Comment by aptonline** [comment]
> Led control and Mic button broke with this release

Does this version fix it?

https://github.com/zhen-zen/YogaSMC/releases/tag/1.4.1

**Comment by simprecicchiani** [comment]
> Does this version fix it?
> 
> https://github.com/zhen-zen/YogaSMC/releases/tag/1.4.1

Yes it does, everything works as expected.

> Hi @simprecicchiani ! I have been using your efi so far I have not seen any inconvenience or bugs, I think it's time to release a separate branch for this version.

I was thinking of releasing it with the next OpenCore update. I was quite happy with ThinkpadAssistant but, from what I understood, it will be soon deprecated. I don't think there is a need for two separate configs.

**Comment by alexdelli** [comment]
Why I can't see the YogaSMC icon?

**Comment by xuandung38** [comment]
OpenCore 0.6.5 released. let's do it <3

**Comment by simprecicchiani** [comment]
> OpenCore 0.6.5 released. let's do it <3

#82

**Comment by mhl221135** [comment]
This thing not working its jumping to balanced and not changing, and what it is performance modes ?)
